Node capacity calculation assert error

If you run the keras cli in test mode with the weights from evaluation sz02/rl01 and with the following sim_config:
```
# module for configuring the simulator
# configuration parameters are loaded and used both when using the simulator via the CLI and via the interface
# all parameters are required, defaults are in comments

inter_arrival_mean: 10.0           # default: 10.0
flow_dr_mean: 1.0                  # default: 1.0
flow_dr_stdev: 0.0                 # default: 0.0
flow_size_shape: 0.1             # default: 0.001
# if deterministic = True, the simulator reinterprets and uses inter_arrival_mean and flow_size_shape as fixed
# deterministic values rather than means of a random distribution
deterministic: True                # default: False
run_duration: 100                  # default: 100
```
You get the following error:
```

Traceback (most recent call last):
  File "/user/coordination-simulation/src/coordsim/simulation/flowsimulator.py", line 234, in process_flow
    assert node_remaining_cap <= node_cap, "Node remaining capacity cannot be more than node capacity!"
AssertionError: Node remaining capacity cannot be more than node capacity!

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/user/coordination-simulation/src/coordsim/simulation/flowsimulator.py", line 121, in pass_flow
    yield self.env.process(self.process_flow(flow, sfc))
AssertionError: Node remaining capacity cannot be more than node capacity!

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/user/coordination-simulation/src/coordsim/simulation/flowsimulator.py", line 95, in init_flow
    yield self.env.process(self.pass_flow(flow, sfc))
AssertionError: Node remaining capacity cannot be more than node capacity!

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/user/rl-scaling-placement/venv/bin/keras", line 11, in <module>
    load_entry_point('rlsp', 'console_scripts', 'keras')()
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/user/rl-scaling-placement/src/rlsp/agents/keras_rl_ddpg.py", line 91, in cli
    testing(agent, env, callbacks, steps, logger, result)
  File "/user/rl-scaling-placement/src/rlsp/agents/keras_rl_ddpg.py", line 364, in testing
    agent.test(env, nb_episodes=10, visualize=True, verbose=1, nb_max_episode_steps=30, callbacks=callbacks)
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/rl/core.py", line 349, in test
    observation, r, d, info = env.step(action)
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/gym/wrappers/time_limit.py", line 15, in step
    observation, reward, done, info = self.env.step(action)
  File "/user/rl-scaling-placement/src/rlsp/envs/scale_and_place_simulated.py", line 142, in step
    current_simulator_state_vectorized, self.current_simulator_state = self.simulator_wrapper.apply(action)
  File "/user/rl-scaling-placement/src/rlsp/envs/simulator_wrapper.py", line 169, in apply
    state = self.simulator.apply(simulator_action)
  File "/user/coordination-simulation/src/siminterface/simulator.py", line 89, in apply
    self.env.run(until=(self.duration * self.run_times))
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/simpy/core.py", line 138, in run
    self.step()
  File "/user/rl-scaling-placement/venv/lib64/python3.6/site-packages/simpy/core.py", line 230, in step
    raise exc
AssertionError: Node remaining capacity cannot be more than node capacity!
```

--- a/src/coordsim/network/flow.py
+++ b/src/coordsim/network/flow.py
@@ -2,6 +2,7 @@
 
 Flow class.
 This identifies the flow and its parameters.
+TODO: Add get/set methods
 
 """
 

--- a/src/coordsim/simulation/flowsimulator.py
+++ b/src/coordsim/simulation/flowsimulator.py
@@ -190,25 +190,25 @@ class FlowSimulator:
         metrics.add_processing_delay(processing_delay)
         flow.end2end_delay += processing_delay
         # Get node capacities
-        node_cap = self.params.network.nodes[flow.current_node_id]["cap"]
-        node_remaining_cap = self.params.network.nodes[flow.current_node_id]["remaining_cap"]
+        node_cap = self.params.network.nodes[current_node_id]["cap"]
+        node_remaining_cap = self.params.network.nodes[current_node_id]["remaining_cap"]
         assert node_remaining_cap >= 0, "Remaining node capacity cannot be less than 0 (zero)!"
         # Metrics: Add active flow to the SF once the flow has begun processing.
         metrics.add_active_flow(flow, current_node_id, current_sf)
         if flow.dr <= node_remaining_cap:
             log.info(
-                "Flow {} started proccessing at sf '{}' at node {}. Time: {}, "
+                "Flow {} started proccessing at sf {} at node {}. Time: {}, "
                 "Processing delay: {}".format(flow.flow_id, current_sf, current_node_id, self.env.now,
                                               processing_delay))
 
-            self.params.network.nodes[flow.current_node_id]["remaining_cap"] -= flow.dr
+            self.params.network.nodes[current_node_id]["remaining_cap"] -= flow.dr
             # Just for the sake of keeping lines small, the node_remaining_cap is updated again.
-            node_remaining_cap = self.params.network.nodes[flow.current_node_id]["remaining_cap"]
+            node_remaining_cap = self.params.network.nodes[current_node_id]["remaining_cap"]
 
             yield self.env.timeout(processing_delay)
             log.info(
-                "Flow {} started departing sf '{}' at node {}."
-                " Time {}".format(flow.flow_id, flow.current_sf, flow.current_node_id, self.env.now))
+                "Flow {} started departing sf {} at node {}."
+                " Time {}".format(flow.flow_id, flow.current_sf, current_node_id, self.env.now))
 
             # Check if flow is currently in last SF, if so, then depart flow.
             if (flow.current_position == len(sfc) - 1):
@@ -225,9 +225,9 @@ class FlowSimulator:
                          .format(flow.flow_id, current_sf, current_node_id, self.env.now))
                 # Remove the active flow from the SF after it departed the SF
                 metrics.remove_active_flow(flow, current_node_id, current_sf)
-            self.params.network.nodes[flow.current_node_id]["remaining_cap"] += flow.dr
+            self.params.network.nodes[current_node_id]["remaining_cap"] += flow.dr
             # Just for the sake of keeping lines small, the node_remaining_cap is updated again.
-            node_remaining_cap = self.params.network.nodes[flow.current_node_id]["remaining_cap"]
+            node_remaining_cap = self.params.network.nodes[current_node_id]["remaining_cap"]
 
             # We assert that remaining capacity must at all times be less than the node capacity so that
             # nodes dont put back more capacity than the node's capacity.

--- a/src/coordsim/simulation/flowsimulator.py
+++ b/src/coordsim/simulation/flowsimulator.py
@@ -208,7 +208,7 @@ class FlowSimulator:
             yield self.env.timeout(processing_delay)
             log.info(
                 "Flow {} started departing sf {} at node {}."
-                " Time {}".format(flow.flow_id, flow.current_sf, current_node_id, self.env.now))
+                " Time {}".format(flow.flow_id, current_sf, current_node_id, self.env.now))
 
             # Check if flow is currently in last SF, if so, then depart flow.
             if (flow.current_position == len(sfc) - 1):

